### PR TITLE
Code Review: Unexpected behaviour when resizing content within Artboards (#13639)

### PR DESCRIPTION
**Pull request by opsGavin**
Code review for Unexpected behaviour when resizing content within Artboards (#13639):

Connect to BohemianCoding/Sketch#13639.

### CODE REVIEW

File: Source/BCGeometry.m [hunk 1]
```diff
@@ -258,107 +267,73 @@ static inline CGFloat linearlyScaleValueWithRangeWithoutConstraint( CGFloat valu
   return value;
 }
 
-static inline CGFloat linearlyScaleValueWithRange( CGFloat value, CGFloat fromSize, CGFloat toSize ) {
```
Comment (opsGavin) [inline Source/BCGeometry.m]: This change is actually in `develop`, but it's been cherry-picked across here. You can ignore all the changes from here down to...

File: Source/BCGeometry.m [hunk 2]
```diff
@@ -382,6 +357,30 @@ CGRect BCRectScaledRelativeToSizeWithConstraints( CGRect rectToScale, CGSize rel
   return rectToScale;
 }
 
+/// Used by BCMinumumSizeForRectWithConstraints to find the constrained size in one dimention.
```
Comment (opsGavin) [inline Source/BCGeometry.m]: ... this point here. ([see](https://github.com/BohemianCoding/BCFoundation/pull/127/files#r117460100))

File: Tests/RectGeometryTests.m [hunk 3]
```diff
@@ -73,16 +73,17 @@ - (void)testMoveRectInRect {
 
 #pragma mark - BCRectScaledRelativeToSizeWithConstraints tests
 // Testing constraints with nothing being sizeable
+// We can't really do this so this should behave as if the min and mid are fixed and the max is flexible
```
Comment (opsGavin) [inline Tests/RectGeometryTests.m]: Likewise - cherry-picked from `develop` - you can ignore changes to here

File: Tests/RectGeometryTests.m [hunk 4]
```diff
@@ -259,6 +255,37 @@ - (void)testResizeWithAllFlexible {
   
 }
 
+#pragma mark - BCMinumumSizeForRectWithConstraints tests
```
Comment (opsGavin) [inline Tests/RectGeometryTests.m]: Changes from this point on are new ([see](https://github.com/BohemianCoding/BCFoundation/pull/127/files#r117460636)).

File: Source/BCGeometry.m [hunk 5]
```diff
@@ -382,6 +357,30 @@ CGRect BCRectScaledRelativeToSizeWithConstraints( CGRect rectToScale, CGSize rel
...
+/// I'm referring to dimentions as width, left, etc make the intension of the method clearer.
```
Comment (pieteromvlee) [inline Source/BCGeometry.m]: `intention` I think

**Comment by pieteromvlee** [comment]
Changed the PR to `develop` since v44.1 just got released and we have merge conflicts, sorry

File: Source/BCGeometry.m (hunk 5 above)
Comment (opsGavin) [inline Source/BCGeometry.m]: Well - you learn something new...

https://www.vocabulary.com/dictionary/intension

> The intension of the word "dog" is the whole concept of dog-ness that the word brings to mind.

5d823411

**Pull request merged**